lib/decors/decorator_definition.rb

```diff
@@ -8,12 +8,7 @@ def define_decorator(decorator_name, decorator_class, mixin: false)
             method_definer = mixin ? :define_method : :define_singleton_method
 
             send(method_definer, decorator_name) do |*params, &blk|
-                if singleton_class?
-                    ObjectSpace.each_object(self).first.send(:extend, ::Decors::MethodAdded::ForwardToSingletonListener)
-                    extend(::Decors::MethodAdded::SingletonListener)
-                else
-                    extend(::Decors::MethodAdded::StandardListener)
-                end
+                extend(::Decors::MethodAdded.listener(self)) unless defined? declared_decorators
```

Comment:
I think `defined? declared_decorators` is not necessary. Using `extend` more than once is a no op afaik.